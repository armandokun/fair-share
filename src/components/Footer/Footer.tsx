import { Container, Link, Typography } from "@mui/material";
import { useTranslation } from "react-i18next";

const Footer = () => {
  const { t } = useTranslation("translation", { keyPrefix: "footer" });

  const repoURL = "https://github.com/armandokun/fair-share";

  return (
    <Container
      disableGutters
      sx={{
        display: "flex",
        justifyContent: "space-between",
      }}
    >
      <Typography>{t("copyright.title")}</Typography>
      <Link href={repoURL}>{t("repository.title")}</Link>
    </Container>
  );
};

export default Footer;
