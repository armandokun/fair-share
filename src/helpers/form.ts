export const getSalaryPercent = (
  salaries: Array<number>,
  rentToPay: number
) => {
  const totalSalary = salaries.reduce(
    (accumulator, currentNumber) => accumulator + currentNumber,
    0
  );

  return (rentToPay / totalSalary) * 100;
};

export const getIndividualPayment = (
  salary: number,
  salaryPercent: number
) => {
  const individualPayment = (salaryPercent / 100) * salary

  return individualPayment.toFixed(2)
};

export const decimalCount = (number: number) => {
  const numberAsString = number.toString();

  if (numberAsString.includes('.')) {
    return numberAsString.split('.')[1].length;
  }

  return 0;
}
