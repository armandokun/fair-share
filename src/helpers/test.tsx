import { I18nextProvider } from "react-i18next";

import i18n from "../translations";

export const withIntl = (children: JSX.Element) => {
  return <I18nextProvider i18n={i18n}>{children}</I18nextProvider>;
};
