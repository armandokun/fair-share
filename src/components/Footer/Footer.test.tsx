import { render, screen } from "@testing-library/react";

import Footer from ".";

import { withIntl } from "../../helpers/test";

describe("<Footer />", () => {
  const renderHelper = (element: JSX.Element) => render(withIntl(element));

  it("renders correctly", () => {
    const { container } = renderHelper(<Footer />);

    expect(container).toMatchSnapshot();
  });

  it("renders footer text", () => {
    renderHelper(<Footer />);

    expect(screen.getByText("Â© No copyrights")).toBeInTheDocument();
  });
});
