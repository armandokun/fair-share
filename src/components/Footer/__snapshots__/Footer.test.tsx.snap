// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-1p4tas-MuiContainer-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
    >
      Â© No copyrights
    </p>
  </div>
</div>
`;
